### DISCUSSION TITLE
Upgrade from version 2.3

### DISCUSSION THREAD

**Discussion by MuchaZ** [RavenSystem/esp-homekit-devices#2311] [Firmware Installation]
I've got Home Accessory Architect v2.3.0 I wanted to update it. In the setup mode I chose "Update all now (HAA OTA v3.0.2)", but nothing happens, tried multiple times. How to upgrade from this pretty old version to a new one?

Also, I'm using Shelly 1 and I'm not able there to enter setup mode other than by holding the button on the device – both physical switch on the wall and trying to turn on WiFi AP by having router turned off (after restarting the power at home) don't work. Any ideas? Here are my settings:
`{"c":{"l":2,"b":[{"g":13,"t":5}], "m":360},"a":[{"t":30,"v":14,"w":4,"b":[{"g":5,"p":0},{"g":5,"t":0},{"g":13}]}]}`

**Comment by xrust83** [comment]
You can't update via OTA update, only via cable. If I'm not mistaken, it's about digital signatures. GitHub's system won't let you update from their server until your bootloader's digital signature is up to date. You need to write a new bootloader file, with new signatures.

**Reply by xrust83** [reply]
I thought that you can try to create a local server to update to the new version, and then go back to update via github.com, or use a cable for simpler

**Reply by MuchaZ** [reply]
Thank you for the explanation.
